lib/mayaUsd/fileio/jobs/readJob.cpp

```diff
@@ -457,17 +449,23 @@ UsdMaya_ReadJob::_DoImport(
     }
 
     if (buildInstances) {
-        MDGModifier dgMod;
+        MDGModifier              deleteMasterMod;
         UsdMayaPrimReaderContext readCtx(&mNewNodeRegistry);
-        for (const auto& master: stage.GetMasters()) {
+        for (const auto& master : usdRootPrim.GetStage()->GetMasters()) {
             const SdfPath masterPath = master.GetPath();
-            MObject masterObject =
-                readCtx.GetMayaNode(masterPath, false);
+            MObject       masterObject = readCtx.GetMayaNode(masterPath, false);
             if (masterObject != MObject::kNullObj) {
-                dgMod.deleteNode(masterObject);
+                MStatus    status;
+                MFnDagNode masterNode(masterObject, &status);
+                if (status) {
+                    while (masterNode.childCount()) {
+                        masterNode.removeChildAt(masterNode.childCount() - 1);
+                    }
```

Comment:
With the special builds we use, we get asserts about incorrect children counts on file->new because the instance count and children count differ. It is not something you will see with a shipped Maya cut.